cni-plugins/sriov-passthrough-cni/plugin/sriov-passthrough-cni

```diff
@@ -23,64 +22,30 @@ function main() {
     esac
 }
 
-function flip_vfs_to_default_drivers() {
-    for file in $(find /sys/devices/ -name *sriov_totalvfs*); do
-        pfroot=$(dirname $file)
-
-        # flip all available VFs to default driver
-        echo 0 > $pfroot/sriov_numvfs
-        cat $file > $pfroot/sriov_numvfs
-    done
-}
-
-function from_cni_args() {
-    local var_name="${1:?}"
-    sed -nE "s/.+?${var_name}=([^;]*?)(;.+?|$)/\1/p" <<< "$CNI_ARGS"
-}
-
 function setup_netns() {
     mkdir -p /var/run/netns
     ln -sfT "$CNI_NETNS" "$CONTAINER_NETNS_LINK"
 }
 
-function setns_sriov_ifs() {
-  local sriov_vfs=( /sys/class/net/*/device/virtfn* )
-  local ifs_arr ifs_name
-  for vf in "${sriov_vfs[@]}"; do
-    ifs_arr=( "$vf"/net/* )
-    for ifs in "${ifs_arr[@]}"; do
-        ifs_name="${ifs%%\/net\/*}"
-        ifs_name="${ifs##*\/}"
-        log "Adding vf: $ifs_name to netns $CNI_CONTAINERID"
-        ip link set "$ifs_name" netns "$CNI_CONTAINERID"
-    done
-  done
+function setns_sriov_pfs() {
+  local pf_array=()
+  local -r sriov_pfs=( $(find /sys/class/net/*/device/sriov_numvfs) )
+  [ "${#sriov_pfs[@]}" -eq 0 ] && log "FATAL: No sriov PFs found" && exit 1
 
-  local sriov_pfs=( /sys/class/net/*/device/sriov_numvfs )
-  local ifs_name
-  for ifs in "${sriov_pfs[@]}"; do
-    ifs_name="${ifs%%/device/*}"
-    ifs_name="${ifs_name##*/}"
-        log "Adding pf: $ifs_name to netns $CNI_CONTAINERID"
-    ip link set "$ifs_name" netns "$CNI_CONTAINERID"
+  for pf in "${sriov_pfs[@]}"; do
+    local pf_name="${pf%%/device/*}"
+    pf_name="${pf_name##*/}"
+    if ip link set "$pf_name" netns "$CNI_CONTAINERID"; then
+      if timeout 5s bash -c "until ip netns exec $CNI_CONTAINERID ip link show $pf_name > /dev/null; do sleep 1; done"; then
+        log "Added $pf_name to netns $CNI_CONTAINERID"
+        pf_array+=("$pf_name")
+        [ "${#pf_array[@]}" -eq "$PF_COUNT" ] && log "Allocated requested number of PFs" && return 0
+      fi
+    fi
```

Comment:
Are we going to need more than one PF per job ? if not we can simplify all this and remove PF_COUNT

Reply:
For migration we might need to move 2, but we can do it with calling the CNI twice (see PR description note please).
We can remove this, but i think it doesn't hurt, because if we will want to make it static 2, instead calling the CNI twice, it will allow to just change PF_COUNT, wdyt ?

Reply:
Since we cannot pass PF_COUNT I don't know how migration test will change it when CNI is called, but also I don't know what is needed to call this cni twi for a specific test.

Reply:
Migration test won't change it, its either that we make it static, or we call two annotations from the job yaml.
It's just to support N allocation, if needed i will remove it, i just don't think it hurts.

Reply:
Ok, good enough.